pages/_app.js

```diff
@@ -1,14 +1,27 @@
 import GlobalStyle from "../styles";
 import { SWRConfig } from "swr";
+import { useState } from "react";
 
 const fetcher = (url) => fetch(url).then((response) => response.json());
 
 export default function App({ Component, pageProps }) {
+  const [bookmark, setBookmarks] = useState([]);
+
+  function handleToggleBookmark(id) {
+    setBookmarks((prev) =>
+      prev.includes(id) ? prev.filter((list) => list !== id) : [...prev, id]
+    );
+  }
+
   return (
     <>
       <SWRConfig value={{ fetcher }}>
         <GlobalStyle />
-        <Component {...pageProps} />
+        <Component
+          {...pageProps}
+          bookmark={bookmark}
```

Comment:
```suggestion
          bookmarks={bookmarks}
```
... and of cause you need to rename the prop in the corresponding component, too